client/gandi.go

```diff
@@ -64,41 +76,34 @@ func (g *Gandi) askGandiToBytes(method, path string, params interface{}) (http.H
 	return resp.Header, content, err
 }
 
-func (g *Gandi) askGandiFromBytes(method, path string, params []byte, recipient interface{}) (http.Header, error) {
-	resp, err := g.doAskGandi(method, path, params, nil)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
-	decoder := json.NewDecoder(resp.Body)
-	decoder.Decode(recipient)
-	return resp.Header, nil
-}
-
-func (g *Gandi) doAskGandi(method, path string, params []byte, extraHeaders [][2]string) (*http.Response, error) {
+func (g *Gandi) doAskGandi(method, path string, p interface{}, extraHeaders [][2]string) (*http.Response, error) {
 	var (
 		err error
 		req *http.Request
 	)
+	params, err := json.Marshal(p)
+	if err != nil {
+		return nil, err
+	}
 	client := &http.Client{}
 	suffix := ""
-	if len(g.sharing_id) != 0 {
-		suffix += "?sharing_id=" + g.sharing_id
+	if len(g.config.SharingID) != 0 {
+		suffix += "?sharing_id=" + g.config.SharingID
 	}
 	if params != nil && string(params) != "null" {
-		req, err = http.NewRequest(method, gandiEndpoint+path+suffix, bytes.NewReader(params))
+		req, err = http.NewRequest(method, g.endpoint+path+suffix, bytes.NewReader(params))
 	} else {
-		req, err = http.NewRequest(method, gandiEndpoint+path+suffix, nil)
+		req, err = http.NewRequest(method, g.endpoint+path+suffix, nil)
 	}
 	if err != nil {
 		return nil, err
 	}
-	req.Header.Add("X-Api-Key", g.apikey)
+	req.Header.Add("Authorization", "Apikey " + g.apikey)
```

Comment:
The most recent gofmt would format that without the spaces around the `+`:

`req.Header.Add("Authorization", "Apikey "+g.apikey)`